var/spack/repos/builtin/packages/slate/package.py

```diff
@@ -114,6 +114,11 @@ def cache_test_sources(self):
         install test subdirectory for use during `spack test run`."""
         self.cache_extra_test_sources(["examples"])
 
+    def mpi_launcher(self):
+        if self.spec["mpi"].satisfies("+pmi") and self.spec["slurm"]:
```

Comment:
This will always resolve to false for an externally provided mpi since they aren't associated with variants or dependencies.

Reply:
I think that worked. Perlmutter is down. The test fails on cori, but the failure happens when a test is launched in srun so these changes should be good.